spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/SpanSubscriptionProvider.java

```diff
@@ -38,7 +38,7 @@
 	final Subscriber<? super T> subscriber;
 	final Context context;
 	final String name;
-	private Tracing tracing;
+	private static volatile Tracing TRACING;
```

Comment:
I would suspect this is not generally good, but could be convinced if there is some assumptions in spring boot. If there are some practice like this, it should be in big comments I think.

Reply:
I have revert this code back, but I still think this can be improved, assume if our program run to 5000qps, `beanFactory.getBean` will run to 50W qps, this will more or less affect performance, this can be avoided.

Reply:
I think understanding what's going on with the lazy lookup is important anyway, and if there are other approaches.

maybe @marcingrzejszczak can shed some light about the pattern here and very many other places about lazy bean lookup so that the perf issues can be examined but not addressed in a risk of lockups. others also can feel free to chip in..